streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -0,0 +1,319 @@
+        if (suspended.containsKey(taskId)) {
+            final T task = suspended.get(taskId);
+            if (task.partitions().equals(partitions)) {
+                suspended.remove(taskId);
+                log.trace("{} resuming suspended {} {}", logPrefix, taskTypeName, taskId);
+                task.resume();
+                transitionToRunning(task);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void transitionToRunning(final T task) {
+        running.put(task.id(), task);
+        for (TopicPartition topicPartition : task.partitions()) {
+            runningByPartition.put(topicPartition, task);
+        }
+        for (TopicPartition topicPartition : task.changelogPartitions()) {
+            runningByPartition.put(topicPartition, task);
+        }
+    }
+
+    T runningTaskFor(final TopicPartition partition) {
+        return runningByPartition.get(partition);
+    }
+
+    Set<TaskId> runningTaskIds() {
+        return running.keySet();
+    }
+
+    Map<TaskId, T> runningTaskMap() {
+        return Collections.unmodifiableMap(running);
+    }
+
+    public String toString(final String indent) {
+        final StringBuilder builder = new StringBuilder();
+        describe(builder, running.values(), indent, "Running:");
+        describe(builder, suspended.values(), indent, "Suspended:");
+        describe(builder, restoring.values(), indent, "Restoring:");
+        describe(builder, created.values(), indent, "New:");
+        return builder.toString();
+    }
+
+    private void describe(final StringBuilder builder,
+                          final Collection<T> tasks,
+                          final String indent,
+                          final String name) {
+        builder.append(indent).append(name);
+        for (final T t : tasks) {
+            builder.append(indent).append(t.toString(indent + "\t\t"));
+        }
+        builder.append("\n");
+    }
+
+    List<AbstractTask> allInitializedTasks() {
+        final List<AbstractTask> tasks = new ArrayList<>();
+        tasks.addAll(running.values());
+        tasks.addAll(suspended.values());
+        tasks.addAll(restoring.values());
+        return tasks;
+    }
+
+    Collection<T> suspended() {
+        return suspended.values();
+    }
+
+    Collection<T> restoring() {
+        return Collections.unmodifiableCollection(restoring.values());
+    }
+
+    Collection<TaskId> allAssignedTaskIds() {
+        final List<TaskId> taskIds = new ArrayList<>();
+        taskIds.addAll(running.keySet());
+        taskIds.addAll(suspended.keySet());
+        taskIds.addAll(restoring.keySet());
+        taskIds.addAll(created.keySet());
+        return taskIds;
+    }
+
+    void clear() {
+        runningByPartition.clear();
+        running.clear();
+        created.clear();
+        suspended.clear();
+        restoredPartitions.clear();
+    }
+
+    Set<TaskId> previousTasks() {
```

Comment:
Yes, it would mean that the logging is incorrect. Though the actual use of `previousTasks`, i.e., by the `StreamPartitionAssignor` would be valid. In any-case, i changed it.